internal/kubectl/kubectl.go

```diff
@@ -186,15 +384,36 @@ func (k Kubectl) runWithOutput(command string, options ...string) ([]byte, error
 }
 
 func (k Kubectl) RolloutRestart(resource string, options ...string) error {
-	command := fmt.Sprintf("kubectl rollout restart %s %s", resource, k.getKubeconfig())
+	arg, cleanup, err := k.getKubeconfig()
+	if err != nil {
+		return err
+	}
+
+	if cleanup != nil {
+		defer func() {
+			if err := cleanup(); err != nil {
+				log.Err(err).Msg("failed to cleanup kubeconfig")
+			}
+		}()
+	}
+
+	command := fmt.Sprintf("kubectl rollout restart %s %s", resource, arg)
 	return k.run(command, options...)
 }
 
 // getKubeconfig function returns either the "--kubeconfig <(echo ...)" if kubeconfig is specified, or empty string of none is given
-func (k Kubectl) getKubeconfig() string {
+func (k Kubectl) getKubeconfig() (string, func() error, error) {
 	if k.Kubeconfig == "" {
-		return ""
-	} else {
-		return fmt.Sprintf("--kubeconfig <(echo '%s')", k.Kubeconfig)
+		return "", nil, nil
+	}
+
+	id := uuid.New()
+	tmpName := fmt.Sprintf("/tmp/k%s", hex.EncodeToString(id[:]))
+	if err := os.WriteFile(tmpName, []byte(k.Kubeconfig), os.ModePerm); err != nil {
+		return "", nil, err
 	}
+
+	cleanup := func() error { return os.Remove(tmpName) }
```

Comment:
```suggestion
cleanup := func() {
    if err := os.Remove(tmpName); err != nil {
        log.Err(err).Msg("failed to cleanup kubeconfig")
    }
}
```

And then do a shoret `defer`: 
```
if cleanup != nil {
		defer cleanup()
	}
```
WDYT?

Reply:
Or even `defer cleanup(doCleanup)` so even the condition could be evaluated in the function itself.